File: 1º Semestre/AEDS-I/Linguagem c/Recursão em java/fatorial.java
import java.util.Scanner;

class fatorial{

    public static int fat(int x){
    if(x==0 || x==1){
        return 1;
    }
    return x * fat(x-1);

    }

    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);

        int n;
        n = scanner.nextInt();
        System.out.println("O fatorial do valor: " + n + " eh " + fat(n));
    }
}

File: 1º Semestre/AEDS-I/Linguagem c/Recursão/fatorial.c
#include <stdio.h>

int fatorial(int x){
    if(x==1 || x==0){
        return 1;
    }
    return x * fatorial(x-1);
    
}   

int main (void){

    int n;
    printf("Digite o valor:\n");
    scanf("%d", &n);
    printf("%d", fatorial(n));

    return 0;
}